serving/resource_model_serving.go

```diff
@@ -24,33 +21,16 @@ func ResourceModelServing() common.Resource {
 			m["name"].ForceNew = true
 			common.MustSchemaPath(m, "config", "served_models").ConflictsWith = []string{"config.served_entities"}
 			common.MustSchemaPath(m, "config", "served_entities").ConflictsWith = []string{"config.served_models"}
-			common.MustSchemaPath(m, "config", "served_models", "scale_to_zero_enabled").Required = false
-			common.MustSchemaPath(m, "config", "served_models", "scale_to_zero_enabled").Optional = true
-			common.MustSchemaPath(m, "config", "served_models", "scale_to_zero_enabled").Default = true
-			common.MustSchemaPath(m, "config", "served_models", "name").Computed = true
-			common.MustSchemaPath(m, "config", "served_models", "workload_type").Default = "CPU"
-			// TODO: `config.served_models.workload_type` should be a `Optional+Computed` field. Also consider this for other similar fields.
-			// In this scenario, if a workspace does not have GPU serving, specifying `workload_type` = 'CPU' will get empty response from API.
-			common.MustSchemaPath(m, "config", "served_models", "workload_type").DiffSuppressFunc = func(k, old, new string, d *schema.ResourceData) bool {
-				return old == "" && new == "CPU"
-			}
 			common.MustSchemaPath(m, "config", "traffic_config").Computed = true
+			common.MustSchemaPath(m, "config", "auto_capture_config", "table_name_prefix").Computed = true
+
+			common.MustSchemaPath(m, "config", "served_models", "name").Computed = true
+			common.MustSchemaPath(m, "config", "served_models", "workload_type").Computed = true
 			common.MustSchemaPath(m, "config", "served_models").Deprecated = "Please use 'config.served_entities' instead of 'config.served_models'."
 
-			common.MustSchemaPath(m, "config", "served_entities", "scale_to_zero_enabled").Required = false
-			common.MustSchemaPath(m, "config", "served_entities", "scale_to_zero_enabled").Optional = true
-			common.MustSchemaPath(m, "config", "served_entities", "scale_to_zero_enabled").Default = false
 			common.MustSchemaPath(m, "config", "served_entities", "name").Computed = true
-			common.MustSchemaPath(m, "config", "served_entities", "workload_size").Optional = true
 			common.MustSchemaPath(m, "config", "served_entities", "workload_size").Computed = true
-			common.MustSchemaPath(m, "config", "served_entities", "workload_type").Optional = true
 			common.MustSchemaPath(m, "config", "served_entities", "workload_type").Computed = true
-			common.MustSchemaPath(m, "config", "served_entities", "workload_type").DiffSuppressFunc = func(k, old, new string, d *schema.ResourceData) bool {
-				return old == "" && new == "CPU"
-			}
-			common.MustSchemaPath(m, "config", "auto_capture_config", "catalog_name").ForceNew = true
-			common.MustSchemaPath(m, "config", "auto_capture_config", "schema_name").ForceNew = true
-			common.MustSchemaPath(m, "config", "auto_capture_config", "table_name_prefix").ForceNew = true
```

Comment:
They can be changed under certain circumstances (for example after disabling and enabling). In the cases it can't, the backend will throw

Reply:
The problem is that if it can't change, then we need to force recreation, and we can't do this if backend throws an error

Reply:
The issue is force recreate doesn't always work in this scenario since we don't delete underlying inference tables when a model serving endpoint is deleted

Reply:
Also, why can't we leave this validation to the backend? If someone tries to apply an invalid update, the apply will simply fail. It would be the same as trying to update the endpoint to an invalid model version for example.

Reply:
Terraform expects providers to be able to figure out whether a resource can be updated or needs to be recreated in the plan itself. There isn't such a thing as an invalid update. Resources are either updated to match the target configuration or destroyed and recreated. This is a significant difference because the underlying ID of the resource may change if the resource is recreated but will not change if not recreated. Again, TF needs to know this at planning time in general in order to properly update downstream resources.

> The issue is force recreate doesn't always work in this scenario since we don't delete underlying inference tables when a model serving endpoint is deleted

Where will this specifically be a problem? Is this an issue if someone tries to create a serving endpoint and the catalog/schema/table prefix have already been used?

Reply:
One of the customer's issues that this PR unblocks is that currently when you don't specify a `table_name_prefix`, the backend will automatically create one for you and send that back. However, because of the `ForceNew` attribute, this would crash loop especially in DABs where no diffs are shown, since it would be unable to force new since the table already exists. We added this line
```
common.MustSchemaPath(m, "config", "auto_capture_config", "table_name_prefix").Computed = true
```
to fix that.

> Terraform expects providers to be able to figure out whether a resource can be updated or needs to be recreated in the plan itself. There isn't such a thing as an invalid update. Resources are either updated to match the target configuration or destroyed and recreated.

Bit confused by this, if a customer specifies an invalid parameter on update (for example updating `minPT` to `-10`), that would be a failed update right?

Reply:
From your description, it does make sense that the `table_name_prefix` is computed.

I'm breaking this down into three scenarios:
* Terraform detects no diff between config and resource state. In this case, the resource will be untouched by TF.
* Terraform detects a diff in some fields but not these fields (or any other ForceNew fields). In this case, Terraform will call the Update method of this resource. The request will include user-specified fields as well as potentially computed fields that were returned by the Read method and put into the state.
* Terraform detects a diff in these fields (or other ForceNew fields). Then, Terraform will delete and recreate the model serving endpoint.

> it would be unable to force new since the table already exists.
Would it be easier for users if either the underlying table was dropped when the model serving endpoint is deleted or the model serving endpoint was able to use an existing auto capture table? All other resources that I can think of support the "create/delete/create" round trip, but it seems like this one does not.

> Bit confused by this, if a customer specifies an invalid parameter on update (for example updating minPT to -10), that would be a failed update right?
Yes, I didn't write that very clearly. An invalid update will be rejected by the control plane (ideally). I was focusing more on the point that we can't write our update method like "try to update, and if it fails, delete and recreate" because that would create a new resource with a new resource ID. The logic to determine whether a resource can be updated or needs to be deleted/recreated has to be determined purely by the Terraform plan.

Reply:
Updated the PR